### PR TITLE
Issue 464

### PR DESCRIPTION
**Pull request by briandowns**
### Well detailed description of the change :

Resolves: #

#

### Type of change:

- [ ] Bug fix

- [ ] New feature

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping

- [ ] Tests have been updated to reflect the changes done within this PR (if applicable).

- [ ] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

**Comment by briandowns** [comment]
Might need some assistance @Jason2605 . All tests and examples are passing locally but not in CI and not sure wht the failure is there. Output seems to be cut off.

**Comment by Jason2605** [comment]
Sorry @briandowns been away for the new year! Thanks for working on this, I haven't forgotten, I'll try to get around to seeing why the tests aren't passing

**Comment by Jason2605** [comment]
Yeah for me running in the debug build im getting a segfault when running the annotation tests, will investigate further

File: src/vm/memory.c [hunk 1]
```diff
@@ -101,7 +101,7 @@ static void blackenObject(DictuVM *vm, Obj *object) {
             ObjClass *klass = (ObjClass *) object;
             grayObject(vm, (Obj *) klass->name);
             grayObject(vm, (Obj *) klass->superclass);
-            grayObject(vm, (Obj *) klass->annotations);
+            grayObject(vm, (Obj *) klass->classAnnotations);
```
Comment (Jason2605) [inline src/vm/memory.c:104]: Doing my own internal testing, this and `methodAnnotations` shouldn't need to be grey'd since they'd be in the functions constant table

File: docs/docs/classes.md [hunk 2]
```diff
@@ -659,33 +658,54 @@ class AnnotatedClass {
...
 class AnnotatedClass {
 
 }
 
-print(AnnotatedClass.annotations); // {"Annotation": "Some extra value!"}
+print(AnnotatedClass.classAnnotations); // {"Annotation": "Some extra value!"}
 ```
 
-Multiple annotations can be supplied to classes.
+Method annotations are available on all class methods except initializers.
```
[the remark this answers is not in the document]
Comment (briandowns) [reply docs/docs/classes.md:695]: Most the thought of class initialization consistency. I'm happy to update it to allow for `init` to have annotations as well.

[the remark this answers is not in the document]
Comment (Jason2605) [reply docs/docs/classes.md:695]: Yeah I think allowing it on constructors could be useful, initial thoughts being it may be useful in the future (potentially?) if we were to create some sort of service container and enable some sort of clever dependency injection or something like that

[the remark this answers is not in the document]
Comment (Jason2605) [reply docs/docs/classes.md:695]: Yeah I think allowing it on constructors could be useful, initial thoughts being it may be useful in the future (potentially?) if we were to create some sort of service container and enable some sort of clever dependency injection or something like that

[the remark this answers is not in the document]
Comment (Jason2605) [reply docs/docs/classes.md:695]: No worries, sorry it took a bit longer! Thanks for the effort on this too

[the remark this answers is not in the document]
Comment (briandowns) [reply docs/docs/classes.md]: Thanks for the review. I think we're all set now.

**Comment by Jason2605** [comment]
Something I've noticed from a "correctness" standpoint is this is technically legal:

```
class Test {
    @Test
    @Test1
    @Test2
    @Test3
}

print(Test.methodAnnotations); // {"__annotatedMethodName": {"Test3": nil, "Test1": nil, "Test": nil, "Test2": nil}}
```

It is an edge case scenario but it would be good if this resulted in a compile time error since there's no method, what do you think?

**Comment by briandowns** [comment]
I "think" we're all set.

**Comment by Jason2605** [comment]
Legend, thank you very much for this!

**Comment by briandowns** [comment]
Thanks! Was a fun challenge. And thanks for the guidance and feedback through the process.
